### PR TITLE
Add SK language translation

### PR DESCRIPTION
**Pull request by ingbrzy**
Opened a new pull request to match changes in new branch MK3_3.11.0

### CODE REVIEW

**Comment by leptun** [comment]
Related to #3172

File: Firmware/language.c
Comment (3d-gussner) [inline Firmware/language.c]: Characters with diacritics like `č` can't be used please change it
```
case LANG_CODE_SK: return _n("Slovencina"); //community Slovak contribution
```

File: lang/lang_en_sk.txt
Comment (3d-gussner) [inline lang/lang_en_sk.txt]: You don't need to shorten `filam.` as `Autozav. filament` will fit in the 18 chars limit.
Or you may be able to `Autozaviest fil.` instead.

File: lang/lang-export.sh
Comment (3d-gussner) [inline lang/lang-export.sh]: I think the English term is `Slovak`

File: lang/lang_en_sk.txt
Comment (3d-gussner) [inline lang/lang_en_sk.txt]: No need to shorten `filam.` as `filament` will fit in the limits of c=20 r=4

File: lang/lang_en_sk.txt
Comment (3d-gussner) [inline lang/lang_en_sk.txt]: No need to shorten `filam.` as `filament` will fit in the limits of c=20 r=4

File: lang/lang_en_sk.txt
Comment (3d-gussner) [inline lang/lang_en_sk.txt]: If you can't translate it please use `\x00` to save some space

File: lang/lang_en_sk.txt
Comment (3d-gussner) [inline lang/lang_en_sk.txt]: If you can't translate it please use `\x00` to save some space

File: lang/lang_en_sk.txt
Comment (3d-gussner) [inline lang/lang_en_sk.txt]: Missing new line at the end!

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: You don't need to shorten `filam.` as `Autozav. filament` will fit in the 18 chars limit.
Or you may be able to `Autozaviest fil.` instead.

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: You don't need to shorten `filam.` as `Autozav. filament` will fit in the 18 chars limit.
Or you may be able to `Autozaviest fil.` instead.

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: No need to shorten `filam.` as `filament` will fit in the limits of c=20 r=4

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: No need to shorten `filam.` as `filament` will fit in the limits of c=20 r=4

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: No need to shorten `filam.` as `filament` will fit in the limits of c=20 r=4

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: No need to shorten `filam.` as `filament` will fit in the limits of c=20 r=4

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: In the po files please use `[μm]` these will be automatically replaced by the `lang-import.sh sk` script.

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: In the po files please use `[μm]` these will be automatically replaced by the `lang-import.sh sk` script.

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: In the po files please use `[μm]` these will be automatically replaced by the `lang-import.sh sk` script.

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: In the po files please use `[μm]` these will be automatically replaced by the `lang-import.sh sk` script.

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: In the po files please use `[μm]` these will be automatically replaced by the `lang-import.sh sk` script.

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: In the po files please use `[μm]` these will be automatically replaced by the `lang-import.sh sk` script.

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: In the po files please use `[μm]` these will be automatically replaced by the `lang-import.sh sk` script.

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: In the po files please use `[μm]` these will be automatically replaced by the `lang-import.sh sk` script.

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: If you can't translate it please leave it blank to save some space

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po]: If you can't translate it please leave it blank to save some space

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [inline lang/po/Firmware_sk.po:1767]: This is an incomplete po file missing the last message, please add ending with a new line.
```
# MSG_PRINTER_IP c=18
#: ultralcd.cpp:2028
msgid "Printer IP Addr:"
msgstr "IP adr. tlaciarne:"

```

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: If you can't translate it please leave it blank to save some space

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po]: If you can't translate it please leave it blank to save some space

File: lang/lang-import.sh
Comment (3d-gussner) [inline lang/lang-import.sh]: Can you please provide a list of non aA-zZ characters in Slovak language and how we can replace them to aA-zZ characters?

This would it make possible to use native chars in po files but being replaced automatically to characters the LCD can display.

File: lang/po/new/sk.po
Comment (3d-gussner) [inline lang/po/new/sk.po:1767]: This is an incomplete po file missing the last message, please add ending with a new line.
```
# MSG_PRINTER_IP c=18
#: ultralcd.cpp:2028
msgid "Printer IP Addr:"
msgstr "IP adr. tlaciarne:"

```

**Review by 3d-gussner — CHANGES_REQUESTED** [review]
The documentation isn't updated yet for FW3.11.0 so you couldn't know.
Please add to [Firmware/config.h](https://github.com/prusa3d/Prusa-Firmware/blob/MK3_3.11.0/Firmware/config.h#L66-L77) your language **behind** Croatian, which should look like this:
```
#define COMMUNITY_LANG_GROUP1_HR // Community Croatian language
#define COMMUNITY_LANG_GROUP1_SK // Community Slovakian language
```

Without a 2nd Maintainer who has to review and approve your translation, will not merge this PR even everything looks very good.
So please if you know someone willing to review/approve and join the task, please let us know.

**Comment by 3d-gussner** [comment]
You can verify your changes:
- change to lang folder
- run `./lang-check.py sk`
  - option `--warn-empty` will show you not translated messages
  - option `--information` will show you all messages
- run `./lang-export.sh sk` to create `/po/ Firmware_sk.po` file from `lang_en_sk.po`
- run `./lang-import.sh sk` to create `po/new/lang_en_sk.txt` and `po/new/output-sk.txt`

I hope that helps.

Thanks again for providing a nearly perfect translation PR and your commitment to support this language.:heart:

**Comment by 3d-gussner** [comment]
@ingbrzy Are you member of https://www.prusaprinters.org/group/slovakia-GQe79R6 ? Maybe you could find there someone willing to join/review/approve/maintain.

**Comment by ingbrzy** [comment]
> @ingbrzy Are you member of https://www.prusaprinters.org/group/slovakia-GQe79R6 ? Maybe you could find there someone willing to join/review/approve/maintain.

@shatter136 can be a reviewer/approver of SK translation

File: lang/lang-import.sh
Comment (leptun) [inline lang/lang-import.sh:356]: I'm curious, how many of these characters are used in the translation overall? I'd assume not all capital letters are used at least

**Comment by ingbrzy** [comment]
> You can verify your changes:
> 
> * change to lang folder
> * run `./lang-check.py sk`
>   
>   * option `--warn-empty` will show you not translated messages
>   * option `--information` will show you all messages
> * run `./lang-export.sh sk` to create `/po/ Firmware_sk.po` file from `lang_en_sk.po`
> * run `./lang-import.sh sk` to create `po/new/lang_en_sk.txt` and `po/new/output-sk.txt`
> 
> I hope that helps.
> 
> Thanks again for providing a nearly perfect translation PR and your commitment to support this language.heart

I have this error when running `./lang-export.sh sk` and `./lang-import.sh sk`
`./lang-export.sh sk`
`config.sh started`
 `Arduino main folder: NG!`
 `Arduino builder: NG!`
 `AVR gcc tools:`
  ` objcopy NG!`
  ` objdump NG!`
 `Output folder: NG!`
 `Objects folder: NG!`
 `Generated elf file: NG!`
 `Generated hex file: NG!`
 `Languages: cz de es fr it pl`
 `Community languages: nl ro hu hr sk `
`config.sh finished with errors!`
`Config NG!`

`./lang-import.sh sk`
`config.sh started`
 `Arduino main folder: NG!`
 `Arduino builder: NG!`
 `AVR gcc tools:`
  ` objcopy NG!`
  ` objdump NG!`
 `Output folder: NG!`
 `Objects folder: NG!`
 `Generated elf file: NG!`
 `Generated hex file: NG!`
 `Languages: cz de es fr it pl`
 `Community languages: nl ro hu hr sk `
`config.sh finished with errors!`
`Config NG!`

could you generate .po files for me?

**Comment by 3d-gussner** [comment]
You have to run `PF-build.sh ` once to create the build files.

**Comment by 3d-gussner** [comment]
Can you contact the reviewer you mentioned if she/he is willing to maintain / review / approve your changes?

Did you try the PF-build.sh hex file on your printer or MK404?

**Comment by 3d-gussner** [comment]
As soon we have the approval, I would ask you to test the language on the printer
- check all menu's
- check all sub menu's
- check toggle some settings
- run the wizard
- run belt test
- print something in the language

Report back or submit change PR if you find a message looking odd.
The Croatian translator found one or two translations that he wanted to update before approve it as tested and good.

**Comment by ingbrzy** [comment]
@shatter136 wants to be a reviewer of SK language.. how he has to approve translation? he is already using my SK translation on his MK3 with 3.10...
Also I am using it on MK3, MK3S, MK3S+ with 3.10 at work.. so its well tested.. :)

**Comment by shatter136** [comment]
Hello. So I confirm SK lang working as ingbrzy says. Do we need to do something specific to approve? Thanks.

**Comment by leptun** [comment]
I think it's fine as long as you confirm that there are no obvious typos or layout problems with the translation. You can approve the changes by submitting an approving review to this pull request (In the Files changed tab, on the right side of the page)

**Comment by leptun** [comment]
I've resolved the merge conflict. Hopefully that didn't cause any issues as this is the first time I'm resolving a merge conflict in github directly

File: lang/lang_en_sk.txt
Comment (3d-gussner) [reply lang/lang_en_sk.txt]: @ingbrzy You already updated the 1st shorten word but you forgot the 2nd one.
`python3 lang-check.py sk` snippet
```
[I]: #MSG_AUTOLOADING_ONLY_IF_FSENS_ON c=20 r=4

 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Autoloading filament|
   2 |available only when |
   3 |filament sensor is  |
   4 |turned on...        |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Autom. zavedenie    |
   2 |filamentu je mozne  |
   3 |len pri zapnutom    |
   4 |filam. senzore...   |
```
but 
```
translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Autom. zavedenie    |
   2 |filamentu je mozne  |
   3 |len pri zapnutom    |
   4 |filamentu senzore...|
```
fits the limits. I am not sure if `filamentu` is the correct 2nd word of course.

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [reply lang/po/Firmware_sk.po]: @ingbrzy You already updated the 1st shorten word but you forgot the 2nd one.
`python3 lang-check.py sk` snippet
```
[I]: #MSG_AUTOLOADING_ONLY_IF_FSENS_ON c=20 r=4

 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Autoloading filament|
   2 |available only when |
   3 |filament sensor is  |
   4 |turned on...        |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Autom. zavedenie    |
   2 |filamentu je mozne  |
   3 |len pri zapnutom    |
   4 |filam. senzore...   |
```
but 
```
translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Autom. zavedenie    |
   2 |filamentu je mozne  |
   3 |len pri zapnutom    |
   4 |filamentu senzore...|
```
fits the limits. I am not sure if `filamentu` is the correct 2nd word of course.

File: lang/po/new/sk.po
Comment (3d-gussner) [reply lang/po/new/sk.po]: You didn't update this one.

File: lang/po/new/sk.po
Comment (3d-gussner) [reply lang/po/new/sk.po]: You didn't update this one.

File: lang/po/new/sk.po
Comment (3d-gussner) [reply lang/po/new/sk.po]: Please update this one

File: lang/po/new/sk.po
Comment (3d-gussner) [reply lang/po/new/sk.po]: Please update this one

File: lang/lang-import.sh
Comment (3d-gussner) [reply lang/lang-import.sh]: Thanks for the conversion.
Did you test it?
- modify one message in `/lang/po/new/sk.po` using all characters
- run `./lang-import.sh sk`
- review this one message in `/lang/po/new/lang_en_sk.txt` if all characters have been replaced corretly?

File: lang/lang_en_sk.txt
Comment (3d-gussner) [reply lang/lang_en_sk.txt]: You are fast :1st_place_medal:

File: lang/po/Firmware_sk.po
Comment (3d-gussner) [reply lang/po/Firmware_sk.po]: Thanks for fast update

File: lang/lang-import.sh
Comment (3d-gussner) [reply lang/lang-import.sh]: @ingbrzy I checkout your repo and branch and if it try to run `./lang-import.sh` it fails and shows in `/lang/po/new/nonscii.txt`
```
10:"POT-Creation-Date: Ne 13. februar 2022, 08:09:46 CET "
11:"PO-Revision-Date: Ne 13. februar 2022, 08:09:46 CET "
```
Looking with VScode at `/lang/po/new/sk.po` it shows that the spaces aren't spaces.

Which OS do you run and how do you edit the po files?

Comment (ingbrzy) [reply lang/lang-import.sh]: after run `./lang-import.sh sk` file `/lang/po/new/lang_en_sk.txt` is not generated but `/lang/po/new/sk_filtered.po`

seems it works fine
`ÁáäČčĎďÉéÍíľĽŇňÓóôÔŕŔŠšŤťÚúÝýŽž`
replaced to
`AaaCcDdEeIilLNnOoOorRSsTtUuYyZz`

Comment (3d-gussner) [reply lang/lang-import.sh]: The problem is that spaces `Ne 13. februar 2022, 08:09:46 CET` are U+00a0 and the script shows these in `nonascii.txt`

Which OS which po edit tool do you use?

Comment (leptun) [reply lang/lang-import.sh]: It might be because of the os locale, not the tool

Comment (ingbrzy) [reply lang/lang-import.sh]: Ubuntu 21.10
gedit 40.1
locale en_US
keyboard SK

Comment (3d-gussner) [reply lang/lang-import.sh]: Modified `lang-import.sh` and then it works
```
#replace UTF-8 'μ' to HD44780 A00 'μ'
 #replace 'μ' with 'A00 ROM μ'
 sed -i 's/\xce\xbc/\\xe4/g' $LNG'_filtered.po'
#replace non-break space with space
 sed -i 's/\xc2\xa0/ /g' $LNG'_filtered.po'
```

Still wonder why your po file contains this char.

Comment (3d-gussner) [reply lang/lang-import.sh]: Found this https://stackoverflow.com/questions/17162851/option-to-display-control-characters-in-gedit/17162916 maybe some how gedit modifies the po file with the non-breaking spaces

**Comment by 3d-gussner** [comment]
@ingbrzy As you pointed out that the `./lang-import.sh` doesn't run the 1st time you clone/copy the Prusa-Firmware I am now working on an improved version.

**Review by shatter136 — APPROVED** [review]

File: lang/lang-import.sh
Comment (3d-gussner) [reply lang/lang-import.sh:356]: Resolve this as it is a question.

File: lang/Add_new_language.md
Comment (DRracer) [inline lang/Add_new_language.md:11]: :smile_cat: 3 words and all of them have/had typos 
- palceholder
- placehodler
- hexadecial

**Review by DRracer — APPROVED** [review]
looks good to me :+1:

File: lang/Add_new_language.md
Comment (3d-gussner) [reply lang/Add_new_language.md:11]: Will update that in next PR and also add a workaround for non-block spaces generated by some editors.

**Review by 3d-gussner — APPROVED** [review]

**Comment by SauerDominik** [comment]
@ingbrzy @shatter136 Hi, can you contact me via email: dominik.sauer@prusa3d.cz, I have a few questions regarding the translations.

**Pull request merged**
